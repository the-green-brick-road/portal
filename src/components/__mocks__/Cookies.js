/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }             from '@mui/material';

function Cookies(props) {

    /* --------- Gather inputs --------- */
    const { color } = props;

    /* -------- Defining theme --------- */
    return (
        <Box style={{ minHeight:'30px',zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> Cookies </Typography>
        </Box>
    );

}

export default Cookies;
