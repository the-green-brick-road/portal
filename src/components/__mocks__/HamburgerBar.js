/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerBar component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerBar(props) {

    /* --------- Gather inputs --------- */
    const { height, isNegative, theme={} }  = props
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }

    return (
        <Box style={{height:height, backgroundColor:color, width: '100%'}}>
            <Typography> HamburgerBar </Typography>
        </Box>
    );

}

export default HamburgerBar;
