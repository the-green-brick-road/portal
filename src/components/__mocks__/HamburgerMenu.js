/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerMenu component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerMenu(props) {

    /* --------- Gather inputs --------- */
    const { top , margin, isNegative, theme={} } = props;
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }
    const menuWidthString = `calc( 100vw - 2 * ${margin})`;

    return (
        <Box style={{ minHeight:'30px', width:menuWidthString, top:top, zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> HamburgerMenu </Typography>
        </Box>
    );

}

export default HamburgerMenu;