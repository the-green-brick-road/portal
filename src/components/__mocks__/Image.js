/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'blue' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Image;
