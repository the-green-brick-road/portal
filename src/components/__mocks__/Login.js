/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Login component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @29 may 2023
# Latest revision: 29 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Login(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'green' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Login;
