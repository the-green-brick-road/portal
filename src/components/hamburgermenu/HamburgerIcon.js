/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Hamburger Icon component
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Button }              from '@mui/material';
import { styled }              from '@mui/system';
import { Menu as MenuIcon }    from '@mui/icons-material';

/* Portal includes */
import { useMenu, useLogging } from '../../providers';

const HamburgerIconBox = styled(Button)(({ col, width, height }) => ({
    right: '0px',
    textAlign: 'center',
    borderWidth: '1px',
    borderStyle: 'solid',
    borderColor: col,
    borderRadius: '0px',
    backgroundColor: 'rgba(255,255,255,0)',
    width,
    height,
    padding: 0,
    minWidth: '40px',
}));

function HamburgerIcon(props) {

    /* --------- Gather inputs --------- */
    const { color, width, height }     = props;
    const { isMenuOpen, setMenuState } = useMenu();
    const { logText }                  = useLogging();
    const componentName = 'HamburgerIcon';

    /* ------- Manage menu click ------- */
    const handleHamburgerMenuClick = () => {

        logText(componentName, 'debug', 'workflow', ' handleHamburgerMenuClick');
        setMenuState(!isMenuOpen)

    };

    /* ----------- Define HTML --------- */
    return (
        <HamburgerIconBox aria-label="open drawer" tabIndex={0} type="button" size="small" onClick={handleHamburgerMenuClick} col={color} height={height} width={width} style={{ backgroundColor:'rgba(255,255,255,0)', borderColor:color, borderStyle:'solid' }}>
            <MenuIcon style={{ width, height: width, color }} />
        </HamburgerIconBox>
    );

}

export default HamburgerIcon;
