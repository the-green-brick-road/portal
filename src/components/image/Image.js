/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @22 may 2023
# Latest revision: 14 june 2023
# ---------------------------------------------------- */

/* React includes */
import { Profiler }              from 'react';

/* Portal includes */
import { useDesign, useLogging } from '../../providers';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { name, style={} }    = props;
    const { images }            = useDesign();
    const { onRender }          = useLogging();
    const componentName = 'Image';

    /* ---- Select image if loaded ----- */
    let data = {}
    if (name in images) { data = images[name] }
    const label_picture = `picture-${name}`

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Profiler id={componentName} onRender={onRender}>
            {(data !== {}) && (
                <picture style={style} data-testid={label_picture}>
                    { Object.entries(data).toReversed().map((item) => {

                        let result;
                        const localType = `image/${item[0]}`;
                        const localSizes = Object.entries(item[1]);
                        let localSrcset = '';
                        for (let i = 0; i < localSizes.length; i += 1) {
                            localSrcset = `${localSrcset} ${localSizes[i][1].img} ${localSizes[i][1].width} ,`;
                        }
                        if (item[0] !== 'default') {
                            result = (<source key={item[0]} srcSet={localSrcset} type={localType} />);
                        }
                        return result;
                    })}
                    <img src={data.default} style={style} alt={name}/>
                </picture>
            )}
        </Profiler>
    );
    /* eslint-enable padded-blocks */

}

export default Image;
