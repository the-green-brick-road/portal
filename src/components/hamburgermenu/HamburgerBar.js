/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Hamburger Bar component
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Local includes */
import HamburgerLogo             from './HamburgerLogo';
import HamburgerStack            from './HamburgerStack';
import HamburgerIcon             from './HamburgerIcon';

function HamburgerBar(props) {

    /* --------- Gather inputs --------- */
    const { height = '20px', isNegative = false, theme={palette:{primary:{main:'#000000'},secondary:{main:'#ffffff'}}} }  = props
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */

    let stackcolor = theme.palette.primary.main;
    if (isNegative) { stackcolor = theme.palette.common.white; }

    /* ----------- Define HTML --------- */
    return (
        <HamburgerStack placeholder="hamburgerbar" id="hamburgerbar" direction="row" alignItems="center" justifyContent="space-between" color={stackcolor} padding="0px" height={height}>
            <HamburgerLogo padding="20px" height={height} name='logo' />
            <HamburgerIcon width="30px" height="30px" color={stackcolor} />
        </HamburgerStack>
    );

}

export default HamburgerBar;
