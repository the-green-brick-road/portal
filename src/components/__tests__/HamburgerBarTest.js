/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Analytics Provider test suite
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Test includes */
import { render, prettyDOM, act, screen, fireEvent } from '@testing-library/react'
import { expect, test}                               from '@jest/globals';

/* Material UI includes */
import { Science }                                   from '@mui/icons-material/';

/* Component under test */
import { default as HamburgerBar }                   from '../../components/hamburgermenu/HamburgerBar';

/* Mocks includes */
/* eslint-disable jest/no-mocks-import */
import { useMenu as mockUseMenu, MenuProvider as MockMenuProvider } from '../../providers/__mocks__/MenuProvider';
import { useLogging as mockUseLogging, LoggingProvider as MockLoggingProvider } from '../../providers/__mocks__/LoggingProvider';
import { default as MockImage }  from '../../components/__mocks__/Image';
jest.mock('../../providers', () => ({
    useLogging: (() => { return mockUseLogging(); }),
    useMenu: (()    => { return mockUseMenu(); }),
}));
jest.mock("../../components", () => ({ Image: (props) => MockImage(props) }));

/* eslint-enable jest/no-mocks-import */

describe("HamburgerBar component" ,() => {

    const entries = [
        {
            "id":     "item1",
            "path":   "/item1",
            "icon":   Science,
            "target": "_top",
        },
        {
            "id": "item2",
            "icon" : Science,
            "subitems": [
                {
                    "id": "subitem21",
                    "path": "/subitem21",
                    "target": "_top",
                },
                {
                    "id": "subitem22",
                    "path": "/subitem22",
                    "target": "_top",
                }
            ],
        }
    ];

    const theme = {
        "palette": {
            "common": {
                "black": "#444444",
                "white": "#555555",
            },
            "primary": {
                "main": "#666666",
                "light": "#666667",
                "dark": "#666668",
            },
            "secondary" : {
                "main": "#777777",
                "light": "#777778",
                "dark": "#777779",
            },
            "disabled" : {
                "main": "#888888",
                "light": "#888889",
                "dark": "#888880",
            },
            "text" : {
                "primary": "#111111",
                "secondary": "#222222",
                "disabled": "#333333",
            },
        },
        "direction": "ltr",
        "typography": {
            "h1": {
                "fontSize": "2.5rem",
                "color": "#111111",
                "textTransform": "uppercase",
                "fontFamily": "Roboto Condensed",
            },
            "h2": {
                "fontSize": "2rem",
                "color": "#222222",
                "textTransform": "lowercase",
                "fontFamily": "Roboto Condensed",
            },
            "body1": {
                "fontSize": "14px",
                "color": "black",
                "fontFamily": "Roboto",
                "position": "relative",
                "textAlign": "justify",
            },
        },
    }

    test('Should display a navigation bar', async () => {

        const view = render(
            <div>
                <MockLoggingProvider>
                    <MockMenuProvider entries={entries}>
                        <HamburgerBar height='84px' isNegative={false} theme={theme}/>
                    </MockMenuProvider>
                </MockLoggingProvider>
            </div>
        );
        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false, highlight: false});
        expect(tree).toMatchSnapshot();

    })

    test('Should display a negative navigation bar', async () => {


        const view = render(
            <div>
                <MockLoggingProvider>
                    <MockMenuProvider entries={entries}>
                        <HamburgerBar height='84px' isNegative={true} theme={theme}/>
                    </MockMenuProvider>
                </MockLoggingProvider>
            </div>
        );
        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false,highlight: false});
        expect(tree).toMatchSnapshot();

    })

    test('Should display a negative navigation event with no attributes', async () => {

        const view = render(
            <div>
                <MockLoggingProvider>
                    <MockMenuProvider entries={entries}>
                        <HamburgerBar />
                    </MockMenuProvider>
                </MockLoggingProvider>
            </div>
        );
        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false,highlight: false});
        expect(tree).toMatchSnapshot();

    })

    test('Should open and close hamburger menu', async () => {

        const state = mockUseMenu();

        let view = null
        await act(async () => { // eslint-disable-line testing-library/no-unnecessary-act

            view = render(
                <div>
                    <MockLoggingProvider>
                        <MockMenuProvider entries={entries}>
                            <HamburgerBar height='84px' isNegative={true} theme={theme}/>
                        </MockMenuProvider>
                    </MockLoggingProvider>
                </div>
            );

        })

        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false,highlight: false});
        expect(tree).toMatchSnapshot();

        await act(async () => {fireEvent.click(screen.getByRole('button', { name: 'open drawer' }))}) // eslint-disable-line testing-library/no-unnecessary-act
        expect(state.setMenuState).toHaveBeenLastCalledWith(true)


    })

})
