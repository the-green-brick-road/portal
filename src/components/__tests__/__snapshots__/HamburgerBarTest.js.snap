// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerBar component Should display a navigation bar 1`] = `
"<body>
  <div>
    <div>
      <div
        class=\\"MuiStack-root css-7k4jhb-MuiStack-root\\"
        id=\\"hamburgerbar\\"
        placeholder=\\"hamburgerbar\\"
      >
        <a
          class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-142vkfw-MuiTypography-root-MuiLink-root\\"
          href=\\"/\\"
        >
          <div
            class=\\"MuiBox-root css-0\\"
            style=\\"height: calc( 84px - 2 * 20px ); vertical-align: middle; top: 20px; background-color: blue;\\"
          />
        </a>
        <button
          aria-label=\\"open drawer\\"
          class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1dwr8jp-MuiButtonBase-root-MuiButton-root\\"
          col=\\"#666666\\"
          height=\\"30px\\"
          style=\\"background-color: rgba(255, 255, 255, 0); border-color: #666666; border-style: solid;\\"
          tabindex=\\"0\\"
          type=\\"button\\"
          width=\\"30px\\"
        >
          <svg
            aria-hidden=\\"true\\"
            class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
            data-testid=\\"MenuIcon\\"
            focusable=\\"false\\"
            style=\\"width: 30px; height: 30px; color: rgb(102, 102, 102);\\"
            viewBox=\\"0 0 24 24\\"
          >
            <path
              d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"
            />
          </svg>
          <span
            class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
          />
        </button>
      </div>
    </div>
  </div>
</body>"
`;

exports[`HamburgerBar component Should display a negative navigation bar 1`] = `
"<body>
  <div>
    <div>
      <div
        class=\\"MuiStack-root css-wm7o92-MuiStack-root\\"
        id=\\"hamburgerbar\\"
        placeholder=\\"hamburgerbar\\"
      >
        <a
          class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-142vkfw-MuiTypography-root-MuiLink-root\\"
          href=\\"/\\"
        >
          <div
            class=\\"MuiBox-root css-0\\"
            style=\\"height: calc( 84px - 2 * 20px ); vertical-align: middle; top: 20px; background-color: blue;\\"
          />
        </a>
        <button
          aria-label=\\"open drawer\\"
          class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1h1aaxo-MuiButtonBase-root-MuiButton-root\\"
          col=\\"#555555\\"
          height=\\"30px\\"
          style=\\"background-color: rgba(255, 255, 255, 0); border-color: #555555; border-style: solid;\\"
          tabindex=\\"0\\"
          type=\\"button\\"
          width=\\"30px\\"
        >
          <svg
            aria-hidden=\\"true\\"
            class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
            data-testid=\\"MenuIcon\\"
            focusable=\\"false\\"
            style=\\"width: 30px; height: 30px; color: rgb(85, 85, 85);\\"
            viewBox=\\"0 0 24 24\\"
          >
            <path
              d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"
            />
          </svg>
          <span
            class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
          />
        </button>
      </div>
    </div>
  </div>
</body>"
`;

exports[`HamburgerBar component Should display a negative navigation event with no attributes 1`] = `
"<body>
  <div>
    <div>
      <div
        class=\\"MuiStack-root css-1kadltl-MuiStack-root\\"
        id=\\"hamburgerbar\\"
        placeholder=\\"hamburgerbar\\"
      >
        <a
          class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1f530oc-MuiTypography-root-MuiLink-root\\"
          href=\\"/\\"
        >
          <div
            class=\\"MuiBox-root css-0\\"
            style=\\"height: calc( 20px - 2 * 20px ); vertical-align: middle; top: 20px; background-color: blue;\\"
          />
        </a>
        <button
          aria-label=\\"open drawer\\"
          class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-v8zjom-MuiButtonBase-root-MuiButton-root\\"
          col=\\"#000000\\"
          height=\\"30px\\"
          style=\\"background-color: rgba(255, 255, 255, 0); border-color: #000000; border-style: solid;\\"
          tabindex=\\"0\\"
          type=\\"button\\"
          width=\\"30px\\"
        >
          <svg
            aria-hidden=\\"true\\"
            class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
            data-testid=\\"MenuIcon\\"
            focusable=\\"false\\"
            style=\\"width: 30px; height: 30px; color: rgb(0, 0, 0);\\"
            viewBox=\\"0 0 24 24\\"
          >
            <path
              d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"
            />
          </svg>
          <span
            class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
          />
        </button>
      </div>
    </div>
  </div>
</body>"
`;

exports[`HamburgerBar component Should open and close hamburger menu 1`] = `
"<body>
  <div>
    <div>
      <div
        class=\\"MuiStack-root css-wm7o92-MuiStack-root\\"
        id=\\"hamburgerbar\\"
        placeholder=\\"hamburgerbar\\"
      >
        <a
          class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-142vkfw-MuiTypography-root-MuiLink-root\\"
          href=\\"/\\"
        >
          <div
            class=\\"MuiBox-root css-0\\"
            style=\\"height: calc( 84px - 2 * 20px ); vertical-align: middle; top: 20px; background-color: blue;\\"
          />
        </a>
        <button
          aria-label=\\"open drawer\\"
          class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1h1aaxo-MuiButtonBase-root-MuiButton-root\\"
          col=\\"#555555\\"
          height=\\"30px\\"
          style=\\"background-color: rgba(255, 255, 255, 0); border-color: #555555; border-style: solid;\\"
          tabindex=\\"0\\"
          type=\\"button\\"
          width=\\"30px\\"
        >
          <svg
            aria-hidden=\\"true\\"
            class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
            data-testid=\\"MenuIcon\\"
            focusable=\\"false\\"
            style=\\"width: 30px; height: 30px; color: rgb(85, 85, 85);\\"
            viewBox=\\"0 0 24 24\\"
          >
            <path
              d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"
            />
          </svg>
          <span
            class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
          />
        </button>
      </div>
    </div>
  </div>
</body>"
`;
